lib/IRGen/GenClangDecl.cpp

```diff
@@ -333,3 +339,15 @@ void IRGenModule::finalizeClangCodeGen() {
   ClangCodeGen->HandleTranslationUnit(
       *const_cast<clang::ASTContext *>(ClangASTContext));
 }
+
+void IRGenModule::ensureImplicitCXXDestructorBodyIsDefined(
+    clang::CXXDestructorDecl *destructor) {
+  if (!destructor->isUserProvided() &&
+      !destructor->doesThisDeclarationHaveABody()) {
+    assert(!destructor->isDeleted() &&
+           "Swift cannot handle a type with no known destructor.");
+    // Make sure we define the destructor so we have something to call.
+    auto &sema = Context.getClangModuleLoader()->getClangSema();
+    sema.DefineImplicitDestructor(clang::SourceLocation(), destructor);
+  }
```

Comment:
I think that early returns make this simpler:

```suggestion
  if (destructor->isUserProvided()) return;
  if (destructor->doesThisDeclarationHaveABody()) return;

  assert(!destructor->isDeleted() &&
         "Swift cannot handle a type with no known destructor.");

  // Make sure we define the destructor so we have something to call.
  auto &sema = Context.getClangModuleLoader()->getClangSema();
  sema.DefineImplicitDestructor(clang::SourceLocation(), destructor);
```